app/models/course/survey/answer_option.rb

```diff
@@ -3,4 +3,13 @@ class Course::Survey::AnswerOption < ActiveRecord::Base
   belongs_to :answer, inverse_of: :options
   belongs_to :question_option, class_name: Course::Survey::QuestionOption.name,
                                inverse_of: :answer_options
+
+  validate :question_option_belongs_to_question
+
+  private
+
+  def question_option_belongs_to_question
+    errors.add(:question_option_id, :invalid) \
+      unless answer.question.option_ids.include?(question_option_id)
```

Comment:
Actually would the N+1 would be easier to solve if we move this validation into `survey/answer`?

Intuitively I would have filtered the invalid options in the params, but that would entail placing some business logic into the controller. I'm now wondering where would be an appropriate place to place this. 

Tradeoffs off the top of my head: the current method results in an entire rollback if any of the option_ids are incorrect, while filtering out incorrect `option_ids` at the controller might result in a slightly more manageable way.